bin/reth/src/cli/ext.rs

```diff
@@ -101,17 +101,33 @@ pub trait RethNodeCommandConfig: fmt::Debug {
         Conf: PayloadBuilderConfig,
         Reth: RethNodeComponents,
     {
-        let payload_generator = BasicPayloadJobGenerator::new(
+        let payload_job_config = BasicPayloadJobGeneratorConfig::default()
+            .interval(conf.interval())
+            .deadline(conf.deadline())
+            .max_payload_tasks(conf.max_payload_tasks())
+            .extradata(conf.extradata_rlp_bytes())
+            .max_gas_limit(conf.max_gas_limit());
+
+        #[cfg(feature = "optimism")]
+        let payload_job_config =
+            payload_job_config.compute_pending_block(conf.compute_pending_block());
```

Comment:
Can you expand on this a bit? Not sure if you mean inlining the construction of the `payload_job_config` in the `PayloadBuilderConfig` trait since that is the unknown `Conf` type so that the `compute_pending_block` feature dependent config can happen there.